### DISCUSSION TITLE
Commit SignOff has wrong email

### DISCUSSION THREAD

**Discussion by DanielHabenicht** [community/community#75161] [Codespaces]
### Select Topic Area

Bug

### Body

As reported in older issues and discussions (see below) the sign-off email used for commits made in codespaces is wrong: 
![image](https://github.com/community/community/assets/13590797/8ab12a29-95ec-47dd-b541-4dcd55786706)

Instead of the one provided by GitHub I would expect my own email address and username. 

Workaround:
```bash
env | grep ^GIT_C
GIT_COMMITTER_NAME=GitHub
GIT_COMMITTER_EMAIL=noreply@github.com
# Workaround
unset GIT_COMMITTER_NAME
unset GIT_COMMITTER_EMAIL
```

See other issues: 
- https://github.com/orgs/community/discussions/68579
- https://github.com/orgs/community/discussions/20925
- https://github.com/orgs/community/discussions/20599

**Comment by bam80** [comment]
Could we prevent GIT_COMMITTER_* vars from exporting into the workspace environment, in the first place?

The vars were noticed in these places:

```
/workspaces/.codespaces/shared/.env:
  12  GIT_COMMITTER_NAME=GitHub
  13: GIT_COMMITTER_EMAIL=noreply@github.com

/workspaces/.codespaces/shared/.env-secrets:
  12  GIT_COMMITTER_NAME=***
  13: GIT_COMMITTER_EMAIL=***

/workspaces/.codespaces/shared/.user-secrets.json:
  1: ...
{"type":"EnvironmentVariable","name":"GIT_COMMITTER_NAME","value":"GitHub"},{"type":"EnvironmentVariable","name":"GIT_COMMITTER_EMAIL","value":"noreply@github.com"},

/workspaces/.codespaces/shared/user-secrets-envs.json:
  1: ...
"GIT_COMMITTER_NAME":"GitHub","GIT_COMMITTER_EMAIL":"noreply@github.com"
```

**Reply by bam80** [reply]
Here how I solved it:
in the default image, the envs are read out from here:

_/etc/profile.d/codespaces.sh_:
```bash
if [ ! -z "$SSH_CONNECTION" ]; then
  while read line
  do
      key=$(echo $line | sed "s/=.*//")
      value=$(echo $line | sed "s/$key=//1")
      decodedValue=$(echo $value | base64 -d)
      export $key="$decodedValue"
  done < /workspaces/.codespaces/shared/.env-secrets
fi
```
So I rebuilt the container with the vars removed from `.env-secrets`, and voila!
Just restarting it should work also.
